dotCMS/src/main/java/com/dotmarketing/filters/CMSFilter.java

```diff
@@ -132,11 +141,40 @@ public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
 		String queryString = request.getQueryString();
 		// if a vanity URL
 		if (iAm == IAm.VANITY_URL) {
-
-			rewrite = VirtualLinksCache.getPathFromCache(host.getHostname() + ":" + ("/".equals(uri) ? "/cmsHomePage" : uri.endsWith("/")?uri.substring(0, uri.length() - 1):uri));
+			VanityUrl vanityUrl = null;
```

Comment:
I would like to refactor all of these by using a AssetHandler interface or so

So each block will be an implementation, and you will do something like

AssetHandler handler = assetHandlerMap.get(iAm);
if (handler.handle (...)) {
   return;
}